client/api.c

```diff
@@ -410,76 +410,67 @@ TDNFCheckUpdates(
 uint32_t
 TDNFClean(
     PTDNF pTdnf,
-    TDNF_CLEANTYPE nCleanType,
-    PTDNF_CLEAN_INFO* ppCleanInfo
+    TDNF_CLEANTYPE nCleanType
     )
 {
     uint32_t dwError = 0;
-    PTDNF_CLEAN_INFO pCleanInfo = NULL;
-    char** ppszReposUsed = NULL;
+    PTDNF_REPO_DATA pRepo = NULL;
 
-    if(!pTdnf || !ppCleanInfo)
+    if(!pTdnf)
     {
         dwError = ERROR_TDNF_INVALID_PARAMETER;
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    //Rest of the clean options will be
-    //supported in the next release
-    if(nCleanType != CLEANTYPE_ALL)
+    if(nCleanType == CLEANTYPE_PLUGINS)
     {
         dwError = ERROR_TDNF_CLEAN_UNSUPPORTED;
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    dwError = TDNFAllocateMemory(
-                  1,
-                  sizeof(TDNF_CLEAN_INFO),
-                  (void**)&pCleanInfo);
-    BAIL_ON_TDNF_ERROR(dwError);
-
-    dwError = TDNFCopyEnabledRepos(pTdnf->pRepos, &pCleanInfo->ppszReposUsed);
-    BAIL_ON_TDNF_ERROR(dwError);
-
-    ppszReposUsed = pCleanInfo->ppszReposUsed;
-    if(nCleanType == CLEANTYPE_ALL)
+    for (pRepo = pTdnf->pRepos; pRepo; pRepo = pRepo->pNext)
     {
-        while(*ppszReposUsed)
+        if (strcmp(pRepo->pszId, CMDLINE_REPO_NAME) == 0)
         {
-            dwError = TDNFRepoRemoveCache(pTdnf,*ppszReposUsed);
+            continue;
+        }
+        pr_info("cleaning %s:", pRepo->pszId);
+        if (nCleanType == CLEANTYPE_METADATA || nCleanType == CLEANTYPE_ALL)
```

Comment:
Instead of enum for `CLEANTYPE_*` let's use bit for values:
```
CLEANTYPE_NONE = 0x0
CLEANTYPE_PACKAGES = 0x1
CLEANTYPE_METADATA = 0x2
CLEANTYPE_DBCACHE = 0x4
CLEANTYPE_PLUGINS = 0x8
CLEANTYPE_EXPIRE_CACHE = 0x10
CLEANTYPE_RPMDB = 0x20
CLEANTYPE_ALL = 0xFF
```
And here in this function:
```
if (nCleanType & CLEANTYPE_METADATA) {
...
}
if (nCleanType & CLEANTYPE_DBCACHE) {
...
}
...
```

This is cleaner and scales well. And no need to compare every time with `CLEANTYPE_ALL`

Reply:
Sure, I can do that. I thought about this before, but thought it's not really worth the effort because we either have just one item or all, never a set of 2 or more.